### ISSUE TITLE
Enable GitHub workflows 

### ISSUE THREAD

**Issue by attilaersek** [CyrielRct/midea_ac_lan#11]
The hacs recommended validation jobs are already configured in this repository just not enabled. It might be a good idea to enable it and maybe also configure some sane and free security checks.

**Comment by Qianli-Ma** [comment]
@CyrielRct Could you add topics to the Repo?

**Comment by CyrielRct** [comment]
Sure it's done. I've done the same as the initial project.

**Comment by CyrielRct** [comment]
@attilaersek I'm not sure how to proceed could you help me with that ? 
It sounds like actions are done but failed as no topic was added. Since I added topics it should be fine now. Anything else ?

**Comment by attilaersek** [comment]
@CyrielRct : Yup, i have an idea what could be the problem! I will open a PR quickly.

**Comment by attilaersek** [comment]
@CyrielRct: though some workflows are a bit outdated, the hacs one was not. Checked the code what this check actually looks for.
I think github is a bit misleading due to the fact that this is a fork. This is how you can create repository tags:

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics

I recommend you to go to the editor, adjust the tags/topics to your taste and retry the action.

**Comment by attilaersek** [comment]
Let me know if it was not enough. I can ignore this check in code until we figure it out.

**Comment by wuwentao** [comment]
maybe I found it later, just follow from original georgezhao2010/midea_ac_lan repo issue and PR info and found it now.

if we planed to continue work with it, maybe I can help to maintain it.
but most of the repo permission/options only can be set by the owner, so maybe I need to fork it and fully control the repo settings.
first of all , I need to confirm with @CyrielRct about it.

and I will try to maintain it with us in future.
most of my current company job works with github/gitlab, and I'm also works as my company github projects maintainer, also use python/shell as program language.

**Comment by wuwentao** [comment]
Hello @CyrielRct 

what't the current status and future plan about this?
I'm using github everyday, so just check it again.

Thanks

**Comment by CyrielRct** [comment]
Hi @wuwentao 
I made this fork for myself originally to merge the pending PRs then I shared the link on the original repo to help people out but I'm clearly not a maintainer and I don't have too much time either.

If you can take on this role I'll continue to help as much as I can but it's totally fine with me and I thank you. 
Let me know when you've got it all set up and I'll close this one.

at this time here are people added to Collaborator status : @Qianli-Ma  @attilaersek  @rokam 

This project is clearly used by many person and need someone who's going to keep it going

**Comment by CyrielRct** [comment]
I see you already done it here : https://github.com/wuwentao/midea_ac_lan

**Comment by wuwentao** [comment]
@CyrielRct 
ok，thank you very much for your confirm. 
I'm using github everyday, as my company job MUST be use it, so I will track all the repo/issue/pr status ASAP.


I have forked it and just debug/config/testing/understanding with it.
in additional, I just use HA and midea_ac_lan, but I  didn't check the souce code , but I have used Python as programming language for more years, so I will try to understand it and follow all the process later.

for release job with error input file name,I have add new github action, it will automatic zip release dir and upload to github release page , all the jobs will be finished by github actions, not our manual action.

I'm also woking with HACS official repo to remove old one and add new fork with it, but it may cost more time, I'm not sure with the process time, once it down, I'll send update.

https://github.com/hacs/integration/issues/3682

I will add Collaborator, thank you for your detail info.

**Comment by daniel-toth-leeder** [comment]
@wuwentao @CyrielRct which repo fork should we try to start using? :) AFAISaw @wuwentao has already initiated the process of removing the unmaintained repo from HACS and add his fork to HACS. Should we use that repo?

Ps: really open if I can be of any help. I'm a dev with GH experience :)

**Comment by rokam** [comment]
@daniel-toth-leeder  lets use @wuwentao fork

**Comment by wuwentao** [comment]
> @wuwentao @CyrielRct which repo fork should we try to start using? :) AFAISaw @wuwentao has already initiated the process of removing the unmaintained repo from HACS and add his fork to HACS. Should we use that repo?
> 
> Ps: really open if I can be of any help. I'm a dev with GH experience :)

Hi @daniel-toth-leeder 

please use my fork to do the future changes, and I will continue maintain it now.
as I can  fully control all the project setting and make changes quickly, I'm using github projects in my daily work.
in additional, I will add more people to collaborator list once they submit more PRs later, they will have permission to merge PR in future.
and NOT keep all the merge PR permission in one people.

Thanks
Wentao

**Comment by CyrielRct** [comment]
I'll close this one today.


Le mer. 15 mai 2024, 12 h 14 a.m., Hello World ***@***.***> a
écrit :
